### ISSUE TITLE
Ytview is playing wrong video

### ISSUE THREAD

**Issue by alexanderepstein** [alexanderepstein/Bash-Snippets#45]
**Issue Label:**
* [x] Bug
* [ ] New feature
* [ ] Enhancement
* [ ] New component

**Description:**
No matter what I search by using ```ytview video title goes here``` when i choose the number to play the video it always plays the wrong video. Sometimes this video is on the list sometimes it is not.

### If its a bug make sure to include this section.
**OS and OS version:**
* [ ] Mac
* [ ] Linux 32 Bit
* [x] Linux 64 Bit
* [ ] Windows 32 Bit
* [ ] Windows 64 Bit

OS Version: Elementary OS 0.4.1

@linyostorovovoltos

**Comment by linyostorovovoltos** [comment]
Am trying to reproduce this bug

**Comment by alexanderepstein** [comment]
Sorry meant to include this you should be able to find the problem somewhere within this

```bash
bash -x ytview 10 hours
+ [[ -d /home/fsociety/.cache/ytview ]]
+ rm -rf /home/fsociety/.cache/ytview/
+ player=
+ configuredClient=
+ currentVersion=1.11.0
+ flag=
+ getConfiguredClient
+ command -v curl
+ configuredClient=curl
+ getConfiguredPlayer
+ [[ linux-gnu == \l\i\n\u\x* ]]
+ command -v vlc
+ player=vlc
+ checkInternet
+ echo -e 'GET http://google.com HTTP/1.0\n\n'
+ nc google.com 80
+ '[' 0 -eq 0 ']'
+ return 0
+ getopts 'uc:s:h*:' option
+ [[ 2 == \0 ]]
+ [[ 10 == \h\e\l\p ]]
+ [[ 10 == \u\p\d\a\t\e ]]
+ [[ '' == \s\e\a\r\c\h ]]
+ [[ '' == \c\h\a\n\n\e\l ]]
++ printf '%s ' 10 hours
+ search='10 hours '
+ searchview
++ echo 10 hours
++ tr ' ' +
+ search=10+hours
+ mkdir -p /home/fsociety/.cache/ytview/searches/10+hours
+ httpGet 'https://www.youtube.com/results?search_query=10+hours'
+ grep Duration
+ case "$configuredClient" in
+ curl -A curl -s 'https://www.youtube.com/results?search_query=10+hours'
+ grep 'watch?v='
+ sed 's/aria-describedby="description-id.*//'
+ sed s/spf-link//
+ sed 's/"//'
+ sed s/title=//
+ sed 's/"//'
+ awk '{print $13 " " $14 " " $15 " " $16 " " $17 " " $18 " " $19 " " $20 " " $21 " " $22 " " $23 " " $24 " " $25 " " $26 " " $27 " " $29}'
+ tail -n 17
+ sed 's/data-sessionlink=itct*.//'
+ sed s/spf-prefetch//
+ sed 's/rel="//'
+ awk '{printf "%s.\t%s\n",NR,$0}'
+ sed 's/"//'
+ httpGet 'https://www.youtube.com/results?search_query=10+hours'
+ case "$configuredClient" in
+ grep 'watch?v='
+ curl -A curl -s 'https://www.youtube.com/results?search_query=10+hours'
+ awk '{print $5}'
+ sed 's/"//'
+ sed s/vve-check//
+ sed 's/href="/https:\/\/www.youtube.com/'
+ sed '/^\s*$/d'
+ awk '{printf "%s.\t%s\n",NR,$0}'
+ sed s/class=yt-uix-tile-link//
+ head -n 20 /home/fsociety/.cache/ytview/searches/10+hours/titles.txt
1.	The impossible game [10 hours]  
2.	Watching It&#39;s Everyday Bro For 10 Hours Straight  
3.	Black Desert Online - [2017] 10 Hours at Elrics (8-12M Silver/Hour)  
4.	10 hours of relaxation music - Sleep Meditation Yoga  
5.	10 Hours of Oddly Satisfying Videos  
6.	DANCE TILL YOU&#39;RE DEAD 10 HOURS  
7.	PPAP Pen Pineapple Apple Pen - 10 HOURS  
8.	Spongebob trap remix [10 hours]  
9.	Campfire &amp; River Night Ambiance 10 Hours | Nature White Noise for Sleep, Studying or 
10.	10 Hours of Walking in NYC wearing a Romper  
11.	Watching Dance Till You&#39;re Dead For 10 Hours  
12.	Night rain on a car - 10 hours video with soothing sounds for relaxation and 
13.	HE-MAN HEYEAYEA SONG FOR 10 HOURS  
14.	I MADE THIS ACCOUNT IN 10 HOURS  
15.	Diggy Diggy Hole - 10 Hours [1080p]  
16.	10 Hours of Soviet Communist Music  
17.	ASMR 10 Hours of Tapping, Crinkle &amp; Trigger Sounds - No Talking Just Sounds  dir="ltr">ASMR
+ read -p 'Choose a video: ' titlenumber
Choose a video: 1
++ grep 1 /home/fsociety/.cache/ytview/searches/10+hours/urls.txt
++ awk '{print $2}'
+ vlc 'https://www.youtube.com/watch?v=wI__53kBBKM' 'https://www.youtube.com/watch?v=51skM6WA-dg' 'https://www.youtube.com/watch?v=qsOUv9EzKsg' 'https://www.youtube.com/watch?v=37BdS5KH9K8' 'https://www.youtube.com/watch?v=AS5CxLCWq-Q' 'https://www.youtube.com/watch?v=guo8CHurCpY' 'https://www.youtube.com/watch?v=eh7lp9umG2I' 'https://www.youtube.com/watch?v=ot50UWYyDDM' 'https://www.youtube.com/watch?v=Sagg08DrO5U&amp;list=PL5aHadniAQ0cv6xhHcp-RxPeNlswk1UOk' 'https://www.youtube.com/watch?v=ov5pxaIbJlM' 'https://www.youtube.com/watch?v=y5zQTmkY7GI'
```
Although I go to play number 1 *The impossible game* I end up playing *MyAbridged - 10 Hours of a Fireplace Burning*

**Comment by linyostorovovoltos** [comment]
Alright, fixed the bug. #50

**Comment by alexanderepstein** [comment]
The fix at #50 now shows 19 titles and at least now the fireplace burning title is number 1. However with the same search when i go to play video number 3 which is now position of impossible game it still plays the fire burning video it appears you are sending more then one youtube link to vlc for some reason. I would think you would only send a single link (if the reason for this is to play a second video after the chosen video I would stay away from that vlc starts throwing errors after the first video is played) even though it will continue. The new output:
```bash
bash -x ./ytview 10 hours
+ [[ -d /home/fsociety/.cache/ytview ]]
+ rm -rf /home/fsociety/.cache/ytview/
+ player=
+ configuredClient=
+ currentVersion=1.11.0
+ flag=
+ getConfiguredClient
+ command -v curl
+ configuredClient=curl
+ getConfiguredPlayer
+ [[ linux-gnu == \l\i\n\u\x* ]]
+ command -v vlc
+ player=vlc
+ checkInternet
+ echo -e 'GET http://google.com HTTP/1.0\n\n'
+ nc google.com 80
+ '[' 0 -eq 0 ']'
+ return 0
+ getopts 'vuc:s:h*:' option
+ [[ 2 == \0 ]]
+ [[ 10 == \h\e\l\p ]]
+ [[ 10 == \u\p\d\a\t\e ]]
+ [[ '' == \s\e\a\r\c\h ]]
+ [[ '' == \c\h\a\n\n\e\l ]]
++ printf '%s ' 10 hours
+ search='10 hours '
+ searchview
++ echo 10 hours
++ tr ' ' +
+ search=10+hours
+ mkdir -p /home/fsociety/.cache/ytview/searches/10+hours
+ httpGet 'https://www.youtube.com/results?search_query=10+hours'
+ grep Duration
+ case "$configuredClient" in
+ curl -A curl -s 'https://www.youtube.com/results?search_query=10+hours'
+ grep 'watch?v='
+ awk '{print $13 " " $14 " " $15 " " $16 " " $17 " " $18 " " $19 " " $20 " " $21 " " $22 " " $23 " " $23}'
+ sed 's/"//g'
+ sed s/rel=//g
+ sed 's/"//'
+ sed 's/aria-describedby="description-id.*//g'
+ sed s/title=//
+ sed 's/data-session-link=itct*.//'
+ sed s/spf-prefetch//g
+ sed s/spf-link//
+ awk '{printf "%s.\t%s\n",NR,$0}'
+ httpGet 'https://www.youtube.com/results?search_query=10+hours'
+ grep 'watch?v='
+ case "$configuredClient" in
+ curl -A curl -s 'https://www.youtube.com/results?search_query=10+hours'
+ awk '{print $5}'
+ sed '/^\s*$/d'
+ sed s/vve-check//
+ awk '{printf "%s.\t%s\n",NR,$0}'
+ sed 's/href="/https:\/\/www.youtube.com/'
+ sed s/class=yt-uix-tile-link//
+ sed 's/"//'
+ head -n 20 /home/fsociety/.cache/ytview/searches/10+hours/titles.txt
1.	10 Hours of a Fireplace Burning  
2.	Fireplace 10 hours full HD  
3.	The impossible game [10 hours]  
4.	Watching It&#39;s Everyday Bro For 10 Hours Straight  
5.	Black Desert Online - [2017] 10 Hours at Elrics (8-12M Silver/Hour) Silver/Hour)
6.	10 hours of relaxation music - Sleep Meditation Yoga  
7.	10 Hours of Oddly Satisfying Videos  
8.	DANCE TILL YOU&#39;RE DEAD 10 HOURS  
9.	Spongebob trap remix [10 hours]  
10.	PPAP Pen Pineapple Apple Pen - 10 HOURS  
11.	Campfire &amp; River Night Ambiance 10 Hours | Nature White Noise Noise
12.	10 Hours of Walking in NYC wearing a Romper  
13.	Watching Dance Till You&#39;re Dead For 10 Hours  
14.	Night rain on a car - 10 hours video with soothing soothing
15.	HE-MAN HEYEAYEA SONG FOR 10 HOURS  
16.	I MADE THIS ACCOUNT IN 10 HOURS  
17.	Diggy Diggy Hole - 10 Hours [1080p]  
18.	10 Hours of Soviet Communist Music  
19.	ASMR 10 Hours of Tapping, Crinkle &amp; Trigger Sounds - No No
+ read -p 'Choose a video: ' titlenumber
Choose a video: 3
++ grep 3 /home/fsociety/.cache/ytview/searches/10+hours/urls.txt
++ awk '{print $2}'
+ vlc 'https://www.youtube.com/watch?v=wI__53kBBKM' 'https://www.youtube.com/watch?v=7WLLx2RLHvk' 'https://www.youtube.com/watch?v=37BdS5KH9K8' 'https://www.youtube.com/watch?v=AS5CxLCWq-Q'
```

**Comment by alexanderepstein** [comment]
Additionally it would be nice if some of the special characters were replaced with what they are supposed to be for example you see ```&amp;``` where it should just be ```&``` you see ```&#39``` when it should be ```'```

**Comment by linyostorovovoltos** [comment]
K, sorry I was away, I'm having to edit the way that it chooses the video from the list.

**Comment by linyostorovovoltos** [comment]
Alright @alexanderepstein fixed it.

**Comment by alexanderepstein** [comment]
Fixed with #50 and #51
